v8-riscv-tools/init_cross_env.sh

```diff
@@ -0,0 +1,140 @@
+#            does undefined behaviors.
+V8_ROOT=$PWD
+RV_HOME=$HOME/opt/riscv
+
+# NOTE: This script supposes you could build a sim build successfully.
+#       which means the depot_tools and other prebuilt binary blobs v8 need were ready.
+
+echo "Now we are going to download and build RISC-V GNU toolchain and QEMU."
+echo "A few repos and files will be downloaded at $PWD,"
+echo "And the toolchain is going to be installed at $RV_HOME ."
+echo "Press Ctrl-C if you do not like these configs, otherwise I will continue:"
+read -p "You have 10s to cancel:" -t 10
+
+# +++++++++++++++++++++++++++++++++++++++++
+# RISC-V GNU Toolchain
+# +++++++++++++++++++++++++++++++++++++++++
+
+mkdir -p $RV_HOME
+
+cd $V8_ROOT/
+git clone https://github.com/riscv/riscv-gnu-toolchain
+pushd riscv-gnu-toolchain
+git submodule update --init --recursive
+./configure --prefix=$RV_HOME
+make linux -j $(nproc) || make linux -j 1
+popd
+
+
+export PATH="$PATH:$RV_HOME/bin"
+echo "export PATH=$PATH:$RV_HOME/bin" >> $HOME/.bashrc
+
+echo "If you hit the 'no-riscv64-compiler-found' error, run this sed command:"
+echo "sed -i 's,riscv64-linux-gnu,riscv64-unknown-linux-gnu,' \\"
+echo "     $V8_ROOT/v8/build/toolchain/linux/BUILD.gn"
+
+cat <<"EOT"
+# Now you can cross build a native d8 for RISC-V:
+
+cd $V8_ROOT/v8
+gn gen out/riscv64.native.debug \
+    --args='is_component_build=false
+    is_debug=true target_cpu="riscv64"
+    v8_target_cpu="riscv64" use_goma=false
+    goma_dir="None"
+    treat_warnings_as_errors=false
+    symbol_level = 0'
+ninja -C out/riscv64.native.debug -j $(nproc)
+
+# Remove obj and gen files that not needed.
+rm -rf out/riscv64.native.debug/obj
+rm -rf out/riscv64.native.debug/gen
+EOT
+
+# +++++++++++++++++++++++++++++++++++++++++
+# RISC-V QEMU
+# +++++++++++++++++++++++++++++++++++++++++
+
+cd $V8_ROOT/
+git clone https://github.com/qemu/qemu.git
+cd qemu
+# NOTE: master branch breaks current fedora image
+git checkout v5.0.0
+git submodule update -r --init -f
+./configure --target-list=riscv64-softmmu && make -j $(nproc)
+
+# optional.
+#sudo make install
+
+###########################################################
+# Deploy the Fedora Developer Rawhide on QEMU/RISCV64
+###########################################################
+
+cd $V8_ROOT/
+wget https://dl.fedoraproject.org/pub/alt/risc-v/repo/virt-builder-images/images/Fedora-Developer-Rawhide-20191123.n.0-fw_payload-uboot-qemu-virt-smode.elf
+wget https://dl.fedoraproject.org/pub/alt/risc-v/repo/virt-builder-images/images/Fedora-Developer-Rawhide-20191123.n.0-sda.raw.xz
+unxz -k Fedora-Developer-Rawhide-20191123.n.0-sda.raw.xz
+
+cat <<"EOTT"
+# NOTE: the default image size is relatively small.
+#       If you want to have a larger disk space in QEMU. run:
+
+sudo apt install libguestfs-tools -y
+truncate -r Fedora-Developer-Rawhide-*.raw expanded.raw
+truncate -s 60G expanded.raw
+
+# FIXME: we may not need sudo to run this three commands.
```

Comment:
Address this FIXME?

Reply:
according to Fedora wiki [1] it is not needed. and I run it failed without sudo. It uses virt-* tools which I have not bisect it yet. I'd dig it more deeper to find the reason.

[1] https://fedoraproject.org/wiki/Architectures/RISC-V/Installing

Reply:
in Ubuntu `sudo` is needed. if remove the `sudo`:
```

supermin: kernel: modpath /lib/modules/4.15.0-112-generic
cp: cannot open '/boot/vmlinuz-4.15.0-112-generic' for reading: Permission denied
supermin: cp -p '/boot/vmlinuz-4.15.0-112-generic' '/var/tmp/.guestfs-1000/appliance.d.v912znam/kernel': command failed, see earlier errors
libguestfs: trace: launch = -1 (error)
virt-resize: error: libguestfs error: /usr/bin/supermin exited with error
status 1, see debug messages above
libguestfs: trace: close
libguestfs: closing guestfs handle 0x55766cdb6ef0 (state 0)
libguestfs: command: run: rm
libguestfs: command: run: \ -rf /tmp/libguestfsdVP8S5
```